core/src/test/java/io/github/tcdl/msb/impl/RequesterImplTest.java

```diff
@@ -81,9 +77,20 @@ public void testPublishWaitForResponses() throws Exception {
     }
 
     @Test
+    public void testPublishWaitForResponsesAck() throws Exception {
+        RequesterImpl<RestPayload> requester = initRequesterForResponsesWith(1, 1000, 800, arg ->  fail());
```

Comment:
well, yeah. But this ticket is about fix when waitForResponses is 0, not 1. I suspect that the test with 1 would pass even without the fix implemented.